AsyncDisplayKit/Private/_ASHierarchyChangeSet.mm

```diff
@@ -484,11 +493,58 @@ - (void)_validateUpdate
   }
 }
 
+#pragma mark - Debugging (Private)
+
 - (NSString *)description
 {
-  return [NSString stringWithFormat:@"<%@ %p: deletedSections=%@, insertedSections=%@, deletedItems=%@, insertedItems=%@>", NSStringFromClass(self.class), self, _deletedSections, _insertedSections, _deleteItemChanges, _insertItemChanges];
+  return ASObjectDescriptionMake(self, [self propertiesForDescription]);
+}
+
+- (NSString *)debugDescription
+{
+  return ASObjectDescriptionMake(self, [self propertiesForDebugDescription]);
+}
+
+- (NSMutableArray<NSDictionary *> *)propertiesForDescription
+{
+  NSMutableArray<NSDictionary *> *result = [NSMutableArray array];
+  if (_originalInsertSectionChanges.count > 0) {
+    [result addObject:@{ @"originalInsertSectionChanges" : [_ASHierarchySectionChange smallDescriptionForSectionChanges:_originalInsertSectionChanges] }];
+  }
+  if (_originalDeleteSectionChanges.count > 0) {
+    [result addObject:@{ @"originalDeleteSectionChanges" : [_ASHierarchySectionChange smallDescriptionForSectionChanges:_originalDeleteSectionChanges] }];
+  }
+  if (_reloadSectionChanges.count > 0) {
+    [result addObject:@{ @"reloadSectionChanges" : [_ASHierarchySectionChange smallDescriptionForSectionChanges:_reloadSectionChanges] }];
+  }
+  if (_originalInsertItemChanges.count > 0) {
+    [result addObject:@{ @"originalInsertItemChanges" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_originalInsertItemChanges] }];
+  }
+  if (_originalDeleteItemChanges.count > 0) {
+    [result addObject:@{ @"originalDeleteItemChanges" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_originalDeleteItemChanges] }];
+  }
+  if (_reloadItemChanges.count > 0) {
+    [result addObject:@{ @"reloadItemChanges" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_reloadItemChanges] }];
+  }
+  if (_insertedSections.count > 0) {
+    [result addObject:@{ @"insertedSections" : [_insertedSections as_smallDescription] }];
+  }
+  if (_deletedSections.count > 0) {
+    [result addObject:@{ @"deletedSections" : [_deletedSections as_smallDescription] }];
+  }
+  if (_insertItemChanges.count > 0) {
+    [result addObject:@{ @"insertItemChanges" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_insertItemChanges] }];
+  }
+  if (_deleteItemChanges.count > 0) {
+    [result addObject:@{ @"deleteItemChanges" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_deleteItemChanges] }];
+  }
```

Comment:
Since this description will be consumed by end users, we should probably aim to show them exactly what they requested from us, and only describe "original" changes. That way they can check what we log against their code directly.

What do you think of reducing it down to just this:

```objectivec
if (_reloadSectionChanges.count > 0) {
  [result addObject:@{ @"reloadSections" : [_ASHierarchySectionChange smallDescriptionForSectionChanges:_reloadSectionChanges] }];
}
if (_reloadItemChanges.count > 0) {
  [result addObject:@{ @"reloadItems" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_reloadItemChanges] }];
}
if (_originalDeleteSectionChanges.count > 0) {
  [result addObject:@{ @"deleteSections" : [_ASHierarchySectionChange smallDescriptionForSectionChanges:_originalDeleteSectionChanges] }];
}
if (_originalDeleteItemChanges.count > 0) {
  [result addObject:@{ @"deleteItems" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_originalDeleteItemChanges] }];
}
if (_originalInsertSectionChanges.count > 0) {
  [result addObject:@{ @"insertSections" : [_ASHierarchySectionChange smallDescriptionForSectionChanges:_originalInsertSectionChanges] }];
}
if (_originalInsertItemChanges.count > 0) {
  [result addObject:@{ @"insertItems" : [_ASHierarchyItemChange smallDescriptionForItemChanges:_originalInsertItemChanges] }];
}
```